api/app/Http/Controllers/Api/AuthController.php

```diff
@@ -77,4 +81,75 @@ public function register(UserRegistrationModel $request)
 
         User::create($account);
     }
+
+    /**
+     * Send a mail for resetting the password.
+     *
+     * @param \App\Http\Requests\Api\PasswordResetModel $request
+     *
+     * @return mixed
+     */
+    public function sendResetMail(PasswordResetModel $request)
+    {
+        $user_email = $request->email;
+        $user = User::where('email', $user_email)->first();
+        if ($user && $this->notSpamming($user, env('PASSWORD_RESET_MINUTES', 30))) {
+            // User exists and had no request < PASSWORD_RESET_MINUTES
+            $uuid_token = Uuid::generate(4);
+            $url = url('/reset/'.$uuid_token);
+            $request->request->add(['url' => $url]);
+            $request->request->add(['name' => $user->name]);
+            $this->sendPasswordResetMail($request);
```

Comment:
Why passing the complete request? If we want to use this in another controller/class we always need to pass a request instead of just the data we need